docs/spec/operators/cli.md

```diff
@@ -10,26 +10,76 @@ different output formats, depending on the gadget and the use case.
 
 Fully qualified name: `operator.cli.fields`
 
-### `mode`
+#### `output`
+
+Determines the output format of the data source. It can be either a single output
+mode for all data sources or an output mode for each data source individually in
+the format `datasource:mode,datasource2:mode`.
+
+The following modes are supported:
+
+- `columns`: This is the default output mode. It displays the output in a
+  tabular format with a column for each field in the data source, unless the
+  field is annotated with `columns.hidden: true`. In addition to hiding columns,
+  you can also customize the columns format by using several annotations like
+  `columns.width`, `columns.alignment`, etc., as explained in TODO.
+- `json`: This mode displays the output in JSON format. Note that the output
+  contains all the fields of the data source, even if they annotated with
+  `columns.hidden: true` as this annotation applies only to the `columns` mode.
+  Depending on the data source type, it may be an array of objects or multiple
+  objects separated by newlines.
+- `jsonpretty`: As the `json` mode, but the output is formatted in a more
+  human-readable way.
+- `yaml`: This mode displays the output in YAML format. Like the `json` mode, it
+  contains all the fields of the data source. YAML entries will be separated by
+  `---` to make it easier to read.
+- `none`: This mode does not display any output.
+- `raw`: This mode displays the raw data output. This is useful when you don't
+  want the output to be formatted in any way, maybe because you want to pipe it
+  to another tool or because it's already formatted in the way you want.
 
-Determines the output format:
-- `json`: Outputs in JSON format.
-- `jsonpretty`: Outputs in pretty-printed JSON format.
-- `columns`: Outputs in a columnar format.
-- `yaml`: Outputs in YAML format.
-- `none`: No output.
-- `raw`: Outputs raw data.
+By default, the CLI operator allows setting the output of each data source in
+all the supported modes. However, you can customize the supported modes of your
+data sources by setting the [supported-output-modes](#clisupported-output-modes)
+annotation.
 
-Fully qualified name: `operator.cli.mode`
+Also, you can change the default output mode for each data source by setting the
+[default-output-mode](#clidefault-output-mode) annotation.
+
+Fully qualified name: `operator.cli.output`
 
 Default: `columns`
+
+## Annotations
+
+### Data Source Annotations
+
+#### `cli.clear-screen-before`
+
+Clear the screen before printing the output of the data source. This is useful
+for data sources that emit periodically the bath of data and you want to clear
+the screen before printing the new data.
+
+This annotation is only applicable to the `columns` and `raw` output modes.
+
+#### `cli.supported-output-modes`
+
+Comma-separated list of output modes supported for the data source. This is
+useful when you want to restrict the output modes for a data source.
+
```

Comment:
I have mixed feelings. I think it's useful internally (an operator could register a new output mode), but I don't see why the gadget developer should worry about it.

Reply:
Makes sense. I removed the references to these annotations in the gadget-devel documentation.